src/notificationmanager.cpp

```diff
@@ -69,8 +70,50 @@ NotificationManager::NotificationManager(QObject* parent)
 
 NotificationManager::~NotificationManager()
 {
+    qDeleteAll(interfaces.values());
     qDeleteAll(m_Groups);
     qDeleteAll(m_unresolvedEvents);
+
+}
+
+void NotificationManager::addModem(QString path)
+{
+    DEBUG() << "NotificationManager::addModem" << path;
+    QOfonoMessageWaiting *mw = new QOfonoMessageWaiting(this);
+    interfaces.insert(path, mw);
+
+    if (!m_messageWaiting) ///TODO: hack to use first added modem, update later to support multi-modems/sims
+	    m_messageWaiting = mw;
```

Comment:
Hm this approach will product problems a) if there are many modems b) if modem is removed. 
A) is a separate multi-sim issue which will need changes to libqofono side as well, for example to pass a modem path so the handling slot knows which modem the signal is for.
b) Well...not likely but possible to have the only modem removal signalled...perhaps someone else could continue from here? I have other tasks waiting :)